### DISCUSSION TITLE
Is there any api that can get the IP of the current device ？as shown in the picture below.

### DISCUSSION THREAD

**Discussion by Pactortester** [mitmproxy/mitmproxy#5238] [Q&A] [answered]
#### Problem Description

Is there any api that can get the IP of the current device ？as shown in the picture below.

<img width="325" alt="image" src="https://user-images.githubusercontent.com/29191106/161388418-36b36ad8-ff00-4c74-aa47-7b34e36b7bbc.png">

**Comment by mhils** [comment, marked as answer]
Yes,

- https://docs.mitmproxy.org/dev/api/events.html -> 
- https://docs.mitmproxy.org/dev/api/mitmproxy/http.html#HTTPFlow -> 
- https://docs.mitmproxy.org/dev/api/mitmproxy/flow.html#Flow.client_conn -> 
- https://docs.mitmproxy.org/dev/api/mitmproxy/connection.html#Client

**Reply by Pactortester** [reply]
thanks, i get it!
